franklin/github/views.py

```diff
@@ -190,3 +196,66 @@ def deploy_hook(request):
 def get_user_repos(request):
     if request.method == 'GET':
         return Response(request.user.details.get_user_repos())
+
+def get_access_token(request):
+    """
+    Tries to get the access token from an OAuth Provider
+    :param request:
+    :param backend:
+    :return:
+    """
+    access_token_url = 'https://github.com/login/oauth/access_token'
+    secret = github_secret
+
+    params = {
+        'client_id': request.data.get('clientId'),
+        'redirect_uri': request.data.get('redirectUri'),
+        'client_secret': secret,
+        'code': request.data.get('code'),
+        'accept': 'json'
+    }
+
+    # Exchange authorization code for access token.
+    r = requests.get(access_token_url, params=params)
+
+    try:
+        access_token = dict(parse_qsl(r.text))['access_token']
+    except KeyError:
+        access_token = 'FAILED'
```

Comment:
We probably should return something like `None` here. Returning a string is just going to require the FE to check for that string. Better to return an error.